CMakeLists.txt

```diff
@@ -956,6 +956,11 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_SYSTEM_PROCESSOR MATCHES "ARM6
   set(BOOTSTRAPPING_MODE "OFF")
 endif()
 
+# Disable bootstrapping when we aren't building SwiftSyntax
+if(NOT SWIFT_BUILD_SWIFT_SYNTAX)
```

Comment:
Can we do that in a safer way? I'm afraid that this can lead to accidentally disabling SwiftCompilerSources in a shipped release-compiler (e.g. by someone removing swift-syntax from a build-preset). This could get mostly unnoticed because it only affects optimizations, i.e. the qualify of generated code.

E.g. by excluding platforms for which we expect to have hosttools (macos, linux, windows).

Reply:
I think I'd expect it to go in the other direction. Fully bring back the BOOTSTRAPPING_MODE "OFF" as an option, then if `bootstrapping=off` we also disable the other parts that depend on Swift. Thoughts?

Reply:
A released compiler without swift-syntax wouldn't support macros, so I don't think it can happen accidentally. That said, my `if(NOT SWIFT_BUILD_SWIFT_SYNTAX)` hack is awful. I'm all for bringing back `BOOTSTRAPPING_MODE "OFF"` to make this explicit.

Reply:
Ah! It turns out that we don't specifically need either my hack or `BOOTSTRAPPING_MODE "OFF"`, since we still have the code that disables SwiftCompilerSources when there is no Swift toolchain, here: https://github.com/swiftlang/swift/blob/main/CMakeLists.txt#L966

I'm going to remove this commit entirely for now, and we can (independently) decide whether to bring back an easier way to turn off bootstrapping and ignore any host Swift toolchain. For now, we'll test this in an environment where there is no swift toolchain.

Reply:
Alright, I'm going to refute my own comment 2 hours later. I went ahead and reverted the change that dropped `bootstrapping=off` support so we can more easily turn off bootstrapping locally for testing. I'm still trying to bring this up in a container with no host Swift just to be sure (and that's probably how we would want to configure CI).

Reply:
> I think I'd expect it to go in the other direction. Fully bring back the BOOTSTRAPPING_MODE "OFF" as an option

No. I intentionally made it to not be turned off because people disabled it in their local development and wondered why some things are not working in the compiler.
SwiftCompilerSources are mandatory, except for the purpose of bootstrapping the compiler on new platforms.